### PR TITLE
Feature/views

### PR DESCRIPTION
**Pull request by Armynerh**
Analysed the wireframes deeply to identify all the necessary elements and how a user would navigate from one page to another
Used methods that have been created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -5,5 +5,6 @@ def index
 
   def show
     @user = User.find(params[:id])
+    @posts = @user.posts
```
Comment (DeliceLydia) [inline app/controllers/users_controller.rb]: - [ ] On the user posts you should use `#recent_post` method to render the user's 3 most recent posts currently you are displaying all posts make sure you are displaying all posts only when you click on the button `'See all Post'`

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -6,5 +6,6 @@ def index
 
   def show
     @post = Post.find(params[:id])
+    @comments = @post.comments
```
Comment (DeliceLydia) [inline app/controllers/posts_controller.rb]: - [ ] Please use `post.recent_comments` method to render 5 most recent comments make sure to see all comments once you are on the post page

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Armynerh, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Users controller `#index` action renders all users to it's corresponding Views index page
- Post show page displays post with all comments
- No linter errors


## Required Changes ♻️
_Check the comments under the review._


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -6,6 +6,6 @@ def index
 
   def show
     @post = Post.find(params[:id])
-    @comments = @post.comments
+    @comments = @post.recent_comments
```
Comment (DeliceLydia) [inline app/controllers/posts_controller.rb]: - [ ] Please make sure to display only `5 recent comments` on the user posts page
Screenshot 2023-11-17 at 17 15 57

Comment (DeliceLydia) [reply app/controllers/posts_controller.rb]: - All comments should be displayed once you are on the post page like this:
Screenshot 2023-11-17 at 17 17 22

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -5,6 +5,6 @@ def index
 
   def show
     @user = User.find(params[:id])
-    @posts = @user.posts
+    @posts = @user.recent_posts
```
Comment (DeliceLydia) [inline app/controllers/users_controller.rb]: - [ ] Please display only `3 recent posts` on the user posts page and remember to add this button `<%= link_to 'See all Post', user_posts_path(@user) %>` that lead you where you can see all posts

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Armynerh, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._




### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply app/controllers/users_controller.rb]: Fixed.

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (Abenezer-Tilahun) [reply app/controllers/users_controller.rb]: Fixed.

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (Abenezer-Tilahun) [reply app/controllers/posts_controller.rb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Armynerh 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.


## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**